types/lib/model.d.ts

```diff
@@ -2235,7 +2235,10 @@ export abstract class Model<T = any, T2 = any> extends Hooks {
    * @param target The model that will be associated with hasOne relationship
    * @param options Options for the association
    */
-  public static hasOne(target: typeof Model, options?: HasOneOptions): HasOne;
+  public static hasOne<M, T extends typeof Model>(
+    this: M, 
```

Comment:
Should probably be `{ new (): M } & typeof Model` but feel free to create a generic such as `ModelCtor<M> = { new (): M } & typeof Mode` to clean this up a bit.

Reply:
Added  export type "ModelCtor"